### ISSUE TITLE
context.TODO should not be used in production code

### ISSUE THREAD

**Issue by breml** [go-critic/go-critic#831]
[`context.TODO`](https://golang.org/pkg/context/#TODO) is a temporary thing and should not end up in production code. Therefore go-critic could warn about the usage of `context.TODO`.

**Before:**

```Go
ctx := context.TODO()
```

**After:**

```Go
ctx := context.Background()
```

**Comment by dgryski** [comment]
Background is not always correct. I think TODO itself clarifies that the code is incomplete and is used to map between code with context support and code without yet support.

**Comment by breml** [comment]
@dgryski I agree, Background is not always correct. My point is more about highlighting the incomplete code by the linter.

**Comment by quasilyte** [comment]
As with some other issues, I'll note that this rule can be expressed with gogrep pattern. :)

https://github.com/quasilyte/go-ruleguard/commit/a9170792170dc36ce7b2ec96dc57e1201117761b

```go
m.Match(`context.TODO()`).Report(`might want to replace context.TODO()`)
```

**Comment by cristaloleg** [comment]
Closing as resolved in ruleguard.
